src/core/message-parser.ts

```diff
@@ -40,12 +40,23 @@ export function parseMessage(inboundMessage: Ably.InboundMessage): Message {
     throw new Ably.ErrorInfo(`received incoming message without clientId`, 50000, 500);
   }
 
-  if (message.data.text === undefined) {
+  if (!message.extras) {
+    throw new Ably.ErrorInfo(`received incoming message without extras`, 50000, 500);
+  }
+
+  // For non-delete messages, text is required
+  if (message.action !== ChatMessageAction.MessageDelete && message.data.text === undefined) {
     throw new Ably.ErrorInfo(`received incoming message without text`, 50000, 500);
   }
 
-  if (!message.extras) {
-    throw new Ably.ErrorInfo(`received incoming message without extras`, 50000, 500);
+  // For non-delete messages, extras.headers is required
```

Comment:
We didn't require that `extras.headers` is always an empty object before. Should we now?

Reply:
We already do the same for `text` so figured it made sense to be consistent